lsp/roslyn_ls.lua

```diff
@@ -0,0 +1,195 @@
+
+            local buffers = vim.lsp.get_buffers_by_client_id(ctx.client_id)
+            for _, buf in ipairs(buffers) do
+                vim.lsp.util._refresh("textDocument/diagnostic", { bufnr = buf })
+            end
+
+            ---NOTE: This is used by rzls.nvim for init
+            vim.api.nvim_exec_autocmds("User", { pattern = "RoslynInitialized", modeline = false })
+        end,
+        ["workspace/_roslyn_projectHasUnresolvedDependencies"] = function()
+            vim.notify("Detected missing dependencies. Run dotnet restore command.", vim.log.levels.ERROR, {
+                title = "roslyn.nvim",
+            })
+            return vim.NIL
+        end,
+        ["workspace/_roslyn_projectNeedsRestore"] = function(_, result, ctx)
+            local client = assert(vim.lsp.get_client_by_id(ctx.client_id))
+
+            ---@diagnostic disable-next-line: param-type-mismatch
+            client:request("workspace/_roslyn_restore", result, function(err, response)
+                if err then
+                    vim.notify(err.message, vim.log.levels.ERROR, { title = "roslyn.nvim" })
+                end
+                if response then
+                    for _, v in ipairs(response) do
+                        vim.notify(v.message, vim.log.levels.INFO, { title = "roslyn.nvim" })
+                    end
+                end
+            end)
+
+            return vim.NIL
+        end,
+        ["razor/provideDynamicFileInfo"] = function(_, _, _)
+            vim.notify(
+                "Razor is not supported.\nPlease use https://github.com/tris203/rzls.nvim",
+                vim.log.levels.WARN,
+                { title = "roslyn.nvim" }
+            )
+            return vim.NIL
+        end,
+    }
+end
+
+---@type vim.lsp.ClientConfig
+return {
+  name = "roslyn_ls",
+  offset_encoding = 'utf-8',
+  ---@param dispatchers vim.lsp.rpc.Dispatchers
+  cmd = function (dispatchers)
+    local pipe_name = "/tmp/422df9c8340645ba8966061884b388aa.sock"
+    vim.system({
+        "roslyn_ls", -- or provide the location of dll manually "dotnet", "Microsoft.CodeAnalysis.LanguageServer.dll",
+        "--logLevel",
+        "Trace",
+        "--extensionLogDirectory", -- this property is required by the server
+        fs.joinpath(uv.os_tmpdir(), 'roslyn_ls/logs'),
+        "--pipe",
+        pipe_name
+      })
+    uv.sleep(1500)
+    return vim.lsp.rpc.connect(pipe_name)(dispatchers)
+  end,
+  filetypes = { "cs" },
+  root_markers = { '.sln', '.csproj' },
+  handlers = roslyn_handlers(),
+  on_attach = function (client, bufnr)
+    -- enable inlay hints if LSP server supports it
+    if client.supports_method("textDocument/inlayHint") then
+      vim.lsp.inlay_hint.enable(true, { bufnr=bufnr })
+    end
+
+    -- enable semantic tokens highligting hints
+    if client.supports_method("textDocument/semanticTokens") then
+      client.server_capabilities.semanticTokensProvider = true
+    end
+
```

Comment:
`nvim-lspconfig` should not set inlay hint or semantic tokens to true/false IMO, so this can just be removed